### PR TITLE
Add syntax highlighting and language server. 

### LINKED ISSUE #1: Initialize the Vim kcl plugin and add some basic functions.

### PR DESCRIPTION
**Pull request by xxmao123**
#### 1. Does this PR affect any open issues?(Y/N) and add issue references (e.g. "fix #123", "re #123".):

- [ ] N
- [x] Y 

fix #1 

#### 2. What is the scope of this PR (e.g. component or file name):
init.lua
lua/lsp/init.lua
lua/lsp/handlers.lua
syntax/kcl.vim
ftplugin/kcl.lua

#### 3. Provide a description of the PR(e.g. more details, effects, motivations or doc link):
Add some basic functions and language server. 

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [x] Other

#### 4. Are there any breaking changes?(Y/N) and describe the breaking changes(e.g. more details, motivations or doc link):

- [x] N
- [ ] Y 

#### 5. Are there test cases for these changes?(Y/N) select and add more details, references or doc links:

- [ ] Unit test
- [ ] Integration test
- [ ] Benchmark (add benchmark stats below)
- [ ] Manual test (add detailed scripts or steps below)
- [x] Other

### CODE REVIEW

**Comment by He1pa** [comment]
Is this support for vim or neovim？

**Comment by xxmao123** [comment]
> Is this support for vim or neovim？

yes, it supports for neovim.

File: syntax/kcl.vim [hunk 1]
```diff
@@ -0,0 +1,79 @@
+" Vim syntax file
+" Language:     KCL
+" Maintainer:   KusionStack
```
Comment (Peefy) [inline syntax/kcl.vim:3]: s/KusionStack/kcl-lang

File: init.lua [hunk 2]
```diff
@@ -0,0 +1,3 @@
+require('lsp')
```
Comment (Peefy) [inline init.lua:1]: Can you update readme to explain how to use and develop this extension?

File: lua/lsp/init.lua [hunk 3]
```diff
@@ -0,0 +1,16 @@
+require("lsp.handlers").setup()
+
+-- Setup language servers.
+function check_start_kcl_lsp()
+  local kcl_lsp_config = {
+    name = 'kcl-cli',
+    cmd = {'kcl-language-server'},
+    filetypes = { 'k' }
```
Comment (starkers) [inline lua/lsp/init.lua:8]: I would name the filetype `kcl`

generally in vim we use bigger names for filetype.. EG `python`, `rust`, `yaml` etc

Then we can associate  `*.k` with `filetype=kcl`

eg

```lua
--- map .k file to filetype "kcl"
vim.filetype.add({
    extension = {
        k = "kcl",
    },
})
```

File: init.lua (hunk 2 above)
Comment (xxmao123) [reply init.lua]: ok.

File: lua/lsp/init.lua (hunk 3 above)
Comment (xxmao123) [reply lua/lsp/init.lua]: Thanks for your suggestion. I have added this part in `lua/basic.lua`.

File: init.lua (hunk 2 above)
Comment (xxmao123) [reply init.lua]: I have updated readme. Please review again.

**Review by Peefy — APPROVED** [review]
LGTM

**Pull request merged**